core/src/main/java/bio/terra/pearl/core/service/publishing/PortalPublishingService.java

```diff
@@ -3,110 +3,118 @@
-import bio.terra.pearl.core.model.publishing.ParticipantDashboardAlertChange;
 import bio.terra.pearl.core.model.publishing.PortalEnvironmentChange;
 import bio.terra.pearl.core.model.publishing.PortalEnvironmentChangeRecord;
 import bio.terra.pearl.core.model.publishing.StudyEnvironmentChange;
-import bio.terra.pearl.core.model.publishing.VersionedConfigChange;
-import bio.terra.pearl.core.model.publishing.VersionedEntityChange;
-import bio.terra.pearl.core.model.site.SiteContent;
+import bio.terra.pearl.core.model.study.Study;
 import bio.terra.pearl.core.model.study.StudyEnvironment;
-import bio.terra.pearl.core.model.survey.Survey;
-import bio.terra.pearl.core.service.CascadeProperty;
 import bio.terra.pearl.core.service.exception.internal.InternalServerException;
-import bio.terra.pearl.core.service.notification.TriggerService;
-import bio.terra.pearl.core.service.notification.email.EmailTemplateService;
 import bio.terra.pearl.core.service.portal.PortalDashboardConfigService;
-import bio.terra.pearl.core.service.portal.PortalEnvironmentConfigService;
-import bio.terra.pearl.core.service.portal.PortalEnvironmentLanguageService;
 import bio.terra.pearl.core.service.portal.PortalEnvironmentService;
-import bio.terra.pearl.core.service.site.SiteContentService;
-import bio.terra.pearl.core.service.survey.SurveyService;
+import bio.terra.pearl.core.service.study.StudyEnvironmentService;
+import bio.terra.pearl.core.service.study.StudyService;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import org.apache.commons.beanutils.PropertyUtils;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.ArrayList;
 import java.util.List;
-import java.util.Optional;
-import java.util.UUID;
 
 /**
  * dedicated service for applying deltas to portal environments
  */
 @Service
 public class PortalPublishingService {
-    private final PortalDiffService portalDiffService;
-    private final PortalEnvironmentService portalEnvironmentService;
-    private final PortalEnvironmentConfigService portalEnvironmentConfigService;
+    private final StudyService studyService;
+    private final PortalEnvironmentService portalEnvService;
     private final PortalEnvironmentChangeRecordDao portalEnvironmentChangeRecordDao;
     private final PortalDashboardConfigService portalDashboardConfigService;
-    private final TriggerService triggerService;
-    private final SurveyService surveyService;
-    private final EmailTemplateService emailTemplateService;
-    private final SiteContentService siteContentService;
-    private final StudyPublishingService studyPublishingService;
-    private final PortalEnvironmentLanguageService portalEnvironmentLanguageService;
+    private final StudyEnvironmentService studyEnvironmentService;
+    private final List<PortalEnvPublishable> portalEnvPublishables;
+    private final List<StudyEnvPublishable> studyEnvPublishables;
```

Comment:
Very nice! Could be cool to have a readme explaining the appraoch - i.e. if you extend this interface, it will _automatically_ be a part of this list & automatically be published